front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts

```diff
@@ -336,3 +337,35 @@ export const interpolateOnTime = <ListValues extends string, Time = number>(
 };
 
 export const isGET = (keyValues: string[]) => keyValues[0] === TIME;
+
+export function offsetTrainDepartureAndArrivalTimes(train: Train, dragOffset = 0) {
+  const firstStop = train.base.stops[0];
+  const lastStop = last(train.base.stops) as Stop;
+  const departure = offsetSeconds(firstStop.time + dragOffset);
+  const arrival = offsetSeconds(lastStop.time + dragOffset);
+  return {
+    id: train.id,
+    labels: train.labels,
+    name: train.name,
+    path: train.path,
+    departure,
+    arrival,
+    speed_limit_tags: train.speed_limit_tags,
+  };
+}
+
+export function offsetSelectedTrainDepartureAndArrivalTimes(
+  trains: Train[],
+  selectedTrainId: number,
+  dragOffset = 0
+) {
+  return trains.map((train) => {
+    const usedOffset =
+      selectedTrainId !== undefined && selectedTrainId === train.id ? dragOffset : 0;
+    return offsetTrainDepartureAndArrivalTimes(train, usedOffset);
+  });
```

Comment:
```suggestion
  return trains.map((train) =>
    selectedTrainId === train.id ? offsetTrainDepartureAndArrivalTimes(train, dragOffset) : train
  );
```

Reply:
actually it’s necessary to call the function for all trains. 
for non selected trains, it’s adding the 2 fields `departure : train.base.stops[0].time` and `arrival: …`

Reply:
Hum actually, it does not work. I didn't understand what the function does